### DISCUSSION TITLE
Nuclei is not seeing the SQLI

### DISCUSSION THREAD

**Discussion by rootpentesting** [projectdiscovery/nuclei#3456] [Q&A] [answered]
Greetings, 
Why is nuclei not being able to detect this simple error based SQLI ? 
on this acunetix vulnerable webpage, only once it was able for me to detect it. 
i also tryed both with all templates and only the generic ones.

![image](https://user-images.githubusercontent.com/68161162/226616256-a3dfe457-5d80-4a08-a4c2-b286b2fdac72.png)

**Comment by ehsandeep** [comment, marked as answer]
@rootpentesting nuclei's detection works based on the template you are running, public templates are not written to work against acunetix test page, instead, you can use [fuzzing-templates](https://github.com/projectdiscovery/fuzzing-templates) for the type of scan you are looking to run.

```console
nuclei -t ~/GitHub/fuzzing-templates/ -u http://testphp.vulnweb.com/listproducts.php?cat=1

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.9.0

		projectdiscovery.io

[INF] Using Nuclei Engine 2.9.0 (latest)
[INF] Using Nuclei Templates 9.4.0 (latest)
[INF] Templates added in last update: 65
[INF] Templates loaded for scan: 17
[INF] Targets loaded for scan: 1
[reflected-xss] [http] [medium] http://testphp.vulnweb.com/listproducts.php?cat=1'"><35041
[sqli-error-based:mysql] [http] [critical] http://testphp.vulnweb.com/listproducts.php?cat=1' [SQL syntax; check the manual that corresponds to your MySQL,Warning: mysql_,check the manual that corresponds to your MySQL server version]
```

**Reply by rootpentesting** [reply]
interesting i will try that, however it did found it some times though, just now it doesn't seem to find it anymore.

**Reply by rootpentesting** [reply]
it worked flawlessly after that, thanks for the quick reply.
